utils/defaults/defaults.go

```diff
@@ -111,8 +111,8 @@ func GetAutoPromotionEnabledOrDefault(rollout *v1alpha1.Rollout) bool {
 }
 
 func GetConsecutiveErrorLimitOrDefault(metric *v1alpha1.Metric) int32 {
-	if metric.ConsecutiveErrorLimit != nil {
-		return *metric.ConsecutiveErrorLimit
+	if &metric.ConsecutiveErrorLimit != nil && metric.ConsecutiveErrorLimit.IntValue() > 0 {
+		return int32(metric.ConsecutiveErrorLimit.IntValue())
```

Comment:
I think this is a bug. Doesn't this logic mean you are not allowing for `consecutiveErrorLimit: 0` ? Why was the check added?

Reply:
That was my mistake, I had a pointer error in `analysis_types.go`. Fixed now.